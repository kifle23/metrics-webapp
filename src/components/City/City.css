.city {
    padding: 1.5rem 1rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .city i {
    position: static;
  }
  
  .info {
    display: flex;
    align-items: center;
    gap: 1rem;
    padding-bottom: 1rem;
  }