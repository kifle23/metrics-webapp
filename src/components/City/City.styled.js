import styled from 'styled-components';

export const CityWrapper = styled.div`
`;
