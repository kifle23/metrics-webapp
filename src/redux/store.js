import { configureStore } from '@reduxjs/toolkit';
import weatherReducer from './weatherSlice';

export default configureStore({
  reducer: {
    data: weatherReducer,
  },
});