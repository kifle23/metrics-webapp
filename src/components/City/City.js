import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import styles from './City.css';
import { fetchWeather } from '../../redux/weather';

const City = (props) => {
  const { title, lat, long } = props;
  const dispatch = useDispatch();

  return (
    <Link onClick={() => dispatch(fetchWeather({ lat, long }))} to="/details" className={styles.city}>
      <span>{title}</span>
      <div className={styles.info}>
        <span>AQI: 1</span>
        <i className="right_arrow" />
      </div>
    </Link>
  );
};

export default City;

City.propTypes = {
  title: PropTypes.string.isRequired,
  lat: PropTypes.string.isRequired,
  long: PropTypes.string.isRequired,
};
