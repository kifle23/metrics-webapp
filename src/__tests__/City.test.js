import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import City from '../components/City/City';

describe('City component test', () => {
  it('Renders City', () => {
    const country = renderer
      .create(
        <Provider store={store}>
          <Router>
            <City title="test" lat="3" long="3" />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(country).toMatchSnapshot();
  });
});
