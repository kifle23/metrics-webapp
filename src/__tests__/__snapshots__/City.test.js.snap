// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component test Renders City 1`] = `
<a
  className="city"
  href="/details"
  onClick={[Function]}
>
  <span>
    test
  </span>
</a>
`;
